File: swift/Sources/AmenClient/Resources/Webhooks.swift
public final class Webhooks {
    let c: AmenClient; init(_ c: AmenClient) { self.c = c }
    public func list() async throws -> [Webhook] { try await c.request(.GET, "/web-hooks/") }
    /// `secretKey` in the response is shown ONLY now — store it in the Keychain / a secret manager immediately.
    public func create(url: String) async throws -> Webhook { try await c.request(.POST, "/web-hooks/", json: ["url": url]) }
    public func delete(_ id: String) async throws { let _: Empty = try await c.request(.DELETE, "/web-hooks/\(id)") }
}

File: flutter/amen_client/lib/src/resources/webhooks.dart
import '../client.dart';
import '../models.dart';

class Webhooks {
  final AmenClient _c;
  Webhooks(this._c);
  Future<List<Webhook>> list() async => (await _c.request('GET', '/web-hooks/') as List).map((w) => Webhook((w as Map).cast())).toList();
  /// `secretKey` in the response is shown ONLY now — store it in a secret manager immediately.
  Future<Webhook> create(String url) async => Webhook((await _c.request('POST', '/web-hooks/', json: {'url': url}) as Map).cast());
  Future<void> delete(String id) => _c.request('DELETE', '/web-hooks/$id');
}
